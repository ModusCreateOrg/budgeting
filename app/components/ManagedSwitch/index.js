// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter, Switch } from 'react-router-dom';
import { actions } from 'modules/location';

class ManagedSwitch extends React.Component<> {
  setLocation = (location): void => {
    this.props.setLocation({
      pathname: location.pathname,
      search: location.search,
      hash: location.hash,
    });
  };

  componentDidMount() {
    this.props.history.listen(location => this.setLocation(location));
    this.setLocation(this.props.history.location);
  }

  render() {
    return <Switch location={this.props.location}>{this.props.children}</Switch>;
  }
}

const mapStateToProps = state => ({
  location: state.location,
});

const mapDispatchToProps = {
  setLocation: actions.setLocation,
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(ManagedSwitch)
);
