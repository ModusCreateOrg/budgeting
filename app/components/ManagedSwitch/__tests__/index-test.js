import React from 'react';
import renderer from 'react-test-renderer';
import ManagedSwitch from '..';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

const ChildComponent = () => <div>working</div>;

// Create mock store
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let store = null;

beforeEach(() => {
  store = mockStore({});
});

it('renders a component', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <ManagedSwitch>
            <ChildComponent />
          </ManagedSwitch>
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('sets location on the store', () => {
  const expectedLocation = { pathname: '/test', search: '?testing=true', hash: '' };

  renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/test', search: 'testing=true' }]}>
        <ManagedSwitch>
          <ChildComponent />
        </ManagedSwitch>
      </MemoryRouter>
    </Provider>
  );

  const action = store.getActions()[0];
  expect(action.type).toBe('location/SET_LOCATION');
  expect(action.location.pathname).toBe(expectedLocation.pathname);
});
