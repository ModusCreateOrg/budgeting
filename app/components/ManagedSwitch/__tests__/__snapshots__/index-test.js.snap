// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<div>
  working
</div>
`;
